### PR TITLE
Check spelling 0.0.22

### PR DESCRIPTION
**Pull request by jsoref**
## Summary of the Pull Request

Upgrades [check-spelling to v0.0.22](https://github.com/check-spelling/check-spelling/releases/tag/v0.0.22)

## Detailed Description of the Pull Request / Additional comments

* refreshes workflow
  * enables dependabot PRs to trigger CI (so that in the future you'll be able to see breaking changes to the dictionary paths)
* refreshes metadata
* built-in handling of `\n`/`\r`/`\t` is removed -- This means that the `patterns/0_*.txt` files can be removed.
* this specific PR includes some shim content, in `allow/check-spelling-0.0.21.txt` -- once it this PR merges, it can be removed on a branch and the next CI will clean out items from `expect.txt` relating to the `\r` stuff and suggest replacement content.
* talking to the bot is enabled for forks (but not the master repository)
* SARIF reporting is enabled for PRs w/in a single repository (not across forks)
* In job reports, there's a summary table (space permitting) linking to instances (this is a poor man's SARIF report)
* When a pattern splits a thing that results in check-spelling finding an unrecognized token, that's reported with a distinct category
* When there are items in expect that not longer match anything but more specific items do (e.g. `microsoft` vs. `Microsoft`), there's now a specific category with help/advice
* Fancier excludes suggestions (excluding directories, file types, ...)
* Refreshed dictionaries
* The comment now links to the job summary (which includes SARIF link if available, the details view, and a generated commit that people can use if they're ok w/ the expect changes and don't want to run perl)

## Validation Steps Performed

1. the branch was developed in https://github.com/check-spelling-sandbox/terminal/actions?query=branch%3Acheck-spelling-0.0.22
2. ensuring compatibility with 0.0.21 was done in https://github.com/check-spelling-sandbox/terminal/pull/3
3. this version has been in development for a year and has quite a few improvements, we've been actively dogfooding it throughout this period 😄

## PR Checklist
- [ ] Closes #xxx
- [ ] Tests added/passed
- [ ] Documentation updated
   - If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
- [ ] Schema updated (if necessary)

### CODE REVIEW

File: .github/actions/spelling/advice.md [hunk 1]
```diff
@@ -6,8 +6,6 @@
 
 By default the command suggestion will generate a file named based on your commit. That's generally ok as long as you add the file to your commit. Someone can reorganize it later.
 
-:warning: The command is written for posix shells. If it doesn't work for you, you can manually _add_ (one word per line) / _remove_ items to `expect.txt` and the `excludes.txt` files.
```
Comment (jsoref) [inline .github/actions/spelling/advice.md:9]: the command should now work on Windows shells

File: .github/actions/spelling/candidate.patterns [hunk 2]
```diff
@@ -483,32 +575,62 @@ customObjectInstantitationMethod
...
+# apache
+a2(?:en|dis)
+
+# weak e-tag
+W/"[^"]+"
 
 # the negative lookahead here is to allow catching 'templatesz' as a misspelling
 # but to otherwise recognize a Windows path with \templates\foo.template or similar:
-\\(?:necessary|r(?:eport|esolve[dr]?|esult)|t(?:arget|emplates?))(?![a-z])
+#\\(?:necessary|r(?:eport|esolve[dr]?|esult)|t(?:arget|emplates?))(?![a-z])
```
Comment (jsoref) [inline .github/actions/spelling/candidate.patterns:614]: this is no longer necessary in 0.0.22 as I removed the built-in `\n`/`\r`/`\t` handling

File: .github/actions/spelling/excludes.txt [hunk 3]
```diff
@@ -41,61 +48,62 @@
...
-\.woff2?$
 \.xcf$
-\.xls
 \.xlsx?$
 \.xpm$
-\.yml$
+\.xz$
 \.zip$
 ^\.github/actions/spelling/
-^\.github/fabricbot.json$
```
Comment (jsoref) [inline .github/actions/spelling/excludes.txt:80]: apparently fabricbot is gone

File: .github/actions/spelling/allow/check-spelling-0.0.21.txt
Comment (jsoref) [inline .github/actions/spelling/allow/check-spelling-0.0.21.txt:1]: This file can be removed once this PR merges, it's here to make the v0.0.21 engine happy. (I meant to include it in the initial PR, but missed a push...)

File: .github/workflows/spelling2.yml [hunk 4]
```diff
@@ -43,8 +66,6 @@ on:
   pull_request_target:
     branches:
     - "**"
-    tags-ignore:
-    - "**"
```
Comment (jsoref) [inline .github/workflows/spelling2.yml:47]: this was never supported by `pull_request_target` as it / `pull_request` never applied to tags.

File: .github/workflows/spelling2.yml [hunk 5]
```diff
@@ -60,46 +81,62 @@ jobs:
       contents: read
       pull-requests: read
       actions: read
+      security-events: write
```
Comment (jsoref) [inline .github/workflows/spelling2.yml:84]: this allows creating `SARIF` reports.

File: .github/workflows/spelling2.yml [hunk 6]
```diff
@@ -60,46 +81,62 @@ jobs:
...
     outputs:
       followup: ${{ steps.spelling.outputs.followup }}
     runs-on: ubuntu-latest
-    if: "contains(github.event_name, 'pull_request') || github.event_name == 'push'"
+    if: ${{ contains(github.event_name, 'pull_request') || github.event_name == 'push' }}
```
Comment (jsoref) [inline .github/workflows/spelling2.yml:88]: this makes the VS Code linter happy -- GitHub itself doesn't care

File: .github/workflows/spelling2.yml [hunk 7]
```diff
@@ -60,46 +81,62 @@ jobs:
...
       # note: If you use only_check_changed_files, you do not want cancel-in-progress
       cancel-in-progress: true
     steps:
     - name: check-spelling
       id: spelling
-      uses: check-spelling/check-spelling@v0.0.21
+      uses: check-spelling/check-spelling@v0.0.22
       with:
-        suppress_push_for_open_pull_request: 1
+        suppress_push_for_open_pull_request: ${{ github.actor != 'dependabot[bot]' && 1 }}
```
Comment (jsoref) [inline .github/workflows/spelling2.yml:98]: This enables dependabot PRs to trigger CI (because they change this file which isn't checked in PRs...)

File: .github/workflows/spelling2.yml [hunk 8]
```diff
@@ -60,46 +81,62 @@ jobs:
...
         checkout: true
         check_file_names: 1
-        spell_check_this: check-spelling/spell-check-this@prerelease
+        spell_check_this: microsoft/terminal@main
```
Comment (jsoref) [inline .github/workflows/spelling2.yml:101]: `spell_check_this` is mostly a bootstrapping problem, at this point it should never be encountered for users as check-spelling has been included for a long time, but it mostly makes more sense to reference the repository itself than to reference spell-check-this...

File: .github/workflows/spelling2.yml [hunk 9]
```diff
@@ -60,46 +81,62 @@ jobs:
...
         post_comment: 0
         use_magic_file: 1
-        extra_dictionary_limit: 10
+        report-timing: 1
+        warnings: bad-regex,binary-file,deprecated-feature,ignored-expect-variant,large-file,limited-references,no-newline-at-eof,noisy-file,non-alpha-in-dictionary,token-is-substring,unexpected-line-ending,whitespace-in-dictionary,minified-file,unsupported-configuration,no-files-to-check
```
Comment (jsoref) [inline .github/workflows/spelling2.yml:105]: you can use this to recategorize errors as warnings

File: .github/workflows/spelling2.yml [hunk 10]
```diff
@@ -60,46 +81,62 @@ jobs:
...
+        experimental_apply_changes_via_bot: ${{ github.repository_owner != 'microsoft' && 1 }}
```
Comment (jsoref) [inline .github/workflows/spelling2.yml:106]: this enables people to talk to check-spelling in PRs in forks (but not in PRs to this repository itself)

File: .github/workflows/spelling2.yml [hunk 11]
```diff
@@ -60,46 +81,62 @@ jobs:
...
+        use_sarif: ${{ (!github.event.pull_request || (github.event.pull_request.head.repo.full_name == github.repository)) && 1 }}
```
Comment (jsoref) [inline .github/workflows/spelling2.yml:107]: this enables SARIF reporting for things except PRs across forks.

File: .github/workflows/spelling2.yml [hunk 12]
```diff
@@ -60,46 +81,62 @@ jobs:
...
+        extra_dictionary_limit: 20
         extra_dictionaries:
-          cspell:software-terms/src/software-terms.txt
-          cspell:python/src/python/python-lib.txt
-          cspell:node/node.txt
-          cspell:cpp/src/stdlib-c.txt
+          cspell:software-terms/dict/softwareTerms.txt
```
Comment (jsoref) [inline .github/workflows/spelling2.yml:110]: some dictionaries have been renamed by the upstream

File: .github/workflows/spelling2.yml [hunk 13]
```diff
@@ -123,12 +160,38 @@ jobs:
...
+
+  update:
+    name: Update PR
+    permissions:
+      contents: write
+      pull-requests: write
+      actions: read
+    runs-on: ubuntu-latest
+    if: ${{
+        github.repository_owner != 'microsoft' &&
```
Comment (jsoref) [inline .github/workflows/spelling2.yml:183]: talking to check-spelling is designed to be used in PRs forks -- not in the main repository (and it only works for PRs within a single repository)

File: build/pipelines/ci.yml [hunk 14]
```diff
@@ -102,7 +102,7 @@ stages:
 
   - ${{ if ne(variables['Build.Reason'], 'PullRequest') }}:
     - stage: CodeIndexer
-      displayName: Github CodeNav Indexer
+      displayName: GitHub CodeNav Indexer
```
Comment (jsoref) [inline build/pipelines/ci.yml:105]: brand

File: dep/CLI11/CLI11.hpp [hunk 15]
```diff
@@ -5471,7 +5471,7 @@ namespace CLI
         bool allow_extras_{ false };
 
         /// If ignore, allow extra arguments in the ini file (ie, don't throw an error). INHERITABLE
-        /// if error error on an extra argument, and if capture feed it to the app
```
Comment (jsoref) [inline dep/CLI11/CLI11.hpp:5474]: this was flagged by duplicate word detection

File: dep/jsoncpp/jsoncpp.cpp [hunk 16]
```diff
@@ -1845,7 +1845,7 @@ bool OurReader::decodeNumber(Token& token, Value& decoded) {
     const auto digit(static_cast<Value::UInt>(c - '0'));
     if (value >= threshold) {
       // We've hit or exceeded the max value divided by 10 (rounded down). If
-      // a) we've only just touched the limit, meaing value == threshold,
+      // a) we've only just touched the limit, meaning value == threshold,
```
Comment (jsoref) [inline dep/jsoncpp/jsoncpp.cpp:1848]: oops, I think this file is excluded by `excludes` -- when I started this work, I worked with a default `excludes` list and let this version of check-spelling regenerate its own excludes -- I only later reapplied most of the excludes.

I'll drop this...

File: dep/jsoncpp/jsoncpp.cpp [hunk 17]
```diff
@@ -2471,7 +2471,7 @@ static std::unique_ptr<T> cloneUnique(const std::unique_ptr<T>& p) {
   return r;
 }
 
-// This is a walkaround to avoid the static initialization of Value::null.
+// This is a workaround to avoid the static initialization of Value::null.
```
Comment (jsoref) [inline dep/jsoncpp/jsoncpp.cpp:2474]: this was a cute thing, but, again, I'll drop it

File: doc/Niksa.md [hunk 18]
```diff
@@ -51,7 +51,7 @@ Will this UI enhancement come to other apps on Windows? Almost certainly not. Th
...
 
-As an aside, @bitcrazed wants us to automate performance tests in some conhost specific way, but I haven't quite figured out a controlled environment to do this in yet. The Windows Engineering System runs performance tests each night that give us a coarse grained way of knowing if we messed something up for the whole operating system, and they technically offer a fine grained way for us to insert our own performance tests... but I just haven't got around to that yet. If you have an idea for a way for us to do this in an automated fashion, I'm all ears.
+As an aside, @bitcrazed wants us to automate performance tests in some conhost specific way, but I haven't quite figured out a controlled environment to do this in yet. The Windows Engineering System runs performance tests each night that give us a coarse-grained way of knowing if we messed something up for the whole operating system, and they technically offer a fine-grained way for us to insert our own performance tests... but I just haven't got around to that yet. If you have an idea for a way for us to do this in an automated fashion, I'm all ears.
```
Comment (jsoref) [inline doc/Niksa.md:54]: spell-check-this's list of additional errors has grown

File: doc/specs/drafts/#2634 - Broadcast Input/#2634 - Broadcast Input.md [hunk 19]
```diff
@@ -290,7 +290,7 @@ though. **I recommend we ignore this for now, and leave this as a follow-up**.
 For reference, refer to the following from iTerm2:
 ![image](https://user-images.githubusercontent.com/2578976/64075757-fa971980-ccee-11e9-9e44-47aaf3bca76c.png)
 
-We don't have a menu bar like on MacOS, but we do have a tab context menu. We
```
Comment (jsoref) [inline doc/specs/drafts/#2634 - Broadcast Input/#2634 - Broadcast Input.md:293]: brand

File: doc/specs/drafts/#3327 - Application Theming/#3327 - Application Theming.md [hunk 20]
```diff
@@ -373,7 +373,7 @@ changes, or the active pane in a tab changes:
       `TabRowControl` to match.
 
 The `tab.cornerRadius` might be a bit trickier to implement. Currently, there's
-not a XAML resource that controls this, nor is this something that's exposed by
+no XAML resource that controls this, nor is this something that's exposed by
```
Comment (jsoref) [inline doc/specs/drafts/#3327 - Application Theming/#3327 - Application Theming.md:376]: I didn't leave it enabled, but there's a pattern that flags `not`/`nor` suggesting `neither`/`nor`. I believe in this case `no` is better.

File: .github/actions/spelling/reject.txt [hunk 21]
```diff
@@ -1,6 +1,7 @@
 ^attache$
 ^attacher$
 ^attachers$
+^bellow$
```
Comment (jsoref) [inline .github/actions/spelling/reject.txt:4]: another archaic word I ran across... -- it's unlikely you'll want to reference stoking fires quite like this.

File: .github/actions/spelling/patterns/patterns.txt [hunk 22]
```diff
@@ -24,70 +23,129 @@ ROY\sG\.\sBIV
...
+# Lorem
+(?:\w|\s|[,.])*\b(?i)(?:amet|consectetur|cursus|dolor|eros|ipsum|lacus|libero|ligula|lorem|magna|neque|nulla|suscipit|tempus)\b(?:\w|\s|[,.])*
+
+# hit-count: 3 file-count: 3
+# mailto urls
+mailto:[-a-zA-Z=;:/?%&0-9+@.]{3,}
 
 # hit-count: 2 file-count: 1
-# latex
-\\(?:n(?:ew|ormal|osub)|r(?:enew)|t(?:able(?:of|)|he|itle))(?=[a-z]+)
```
Comment (jsoref) [inline .github/actions/spelling/patterns/patterns.txt:68]: as check-spelling no longer special cases `\n`/`\r`/`\t`, the workaround can be removed :)

File: .github/actions/spelling/patterns/0_n.txt
Comment (jsoref) [inline .github/actions/spelling/patterns/0_n.txt:1]: as check-spelling no longer special cases `\n`/`\r`/`\t`, the workaround can be removed :)

File: .github/actions/spelling/patterns/0_r.txt
Comment (jsoref) [inline .github/actions/spelling/patterns/0_r.txt:1]: as check-spelling no longer special cases `\n`/`\r`/`\t`, the workaround can be removed :)

File: .github/actions/spelling/patterns/0_t.txt
Comment (jsoref) [inline .github/actions/spelling/patterns/0_t.txt:1]: as check-spelling no longer special cases `\n`/`\r`/`\t`, the workaround can be removed :)

File: .github/actions/spelling/allow/check-spelling-0.0.21.txt
Comment (jsoref) [reply .github/actions/spelling/allow/check-spelling-0.0.21.txt:1]: (GitHub's definition of outdated is buggy, there's a github discussion where people complain about this poor behavior -- this comment is not outdated)

**Comment by DHowett** [comment]
AHH, I'm so excited!

We might-could need to revert the changes under `dep/` and `oss/`. I'll noodle on that.

Thank you for doing this!

File: oss/stb/stb_rect_pack.h [hunk 23]
```diff
@@ -466,7 +466,7 @@ static stbrp__findresult stbrp__skyline_pack_rectangle(stbrp_context *context, i
 
    // insert the new node into the right starting point, and
    // let 'cur' point to the remaining nodes needing to be
-   // stiched back in
+   // stitched back in
```
Comment (lhecker) [inline oss/stb/stb_rect_pack.h:469]: Can you revert the changes to the `/dep` and `/oss` directories? Those are external source files we've vendored.

**Review by lhecker — APPROVED** [review]

**Comment by DHowett** [comment]
Okay, wow, I 100% didn't notice that you had done the oss/ and dep/ things. Dang it github! Force pushes are pushes too!

I'll review and summarily merge

File: .github/actions/spelling/reject.txt (hunk 21 above)
Comment (DHowett) [reply .github/actions/spelling/reject.txt:4]: UGH THANK YOU

**Comment by DHowett** [comment]
Ugh, I shouldnae've merged main.
